app/components/SaveControl/SaveControl.js

```diff
@@ -81,47 +98,59 @@ class SaveControl extends Component {
     );
   }
 
-  render() {
-    const { l } = this.context.i18n;
+  rememberArticleInProgress() {
+    return this.props.rememberArticleState && this.props.rememberArticleState.get('isInProgress');
+  }
 
+  updateArticleInProgress() {
+    return this.props.updateArticleState && this.props.updateArticleState.get('isInProgress');
+  }
+
+  spinner(text) {
+    return (
+      <div>
+        <div style={inlineStyles.left}>
+          {text}
+        </div>
+        <div style={inlineStyles.right}>
+          <Spinner size={23} />
+        </div>
+      </div>
+    );
+  }
+
+  render() {
     if (this.props.article.isEmpty() &&
         (!this.state.currentArticle || this.state.currentArticle.isEmpty())) {
       return null;
     }
 
-    if (this.props.article.get('state')) {
+    if (this.props.article.get('id')) {
       this.saveOnUnload(this.props.article.get('url'));
     }
 
     return (
       <div style={inlineStyles.container}>
         <TextField
-          defaultValue={this.props.article.get('url') || this.state.currentArticle.get('url')}
-          disabled
-          floatingLabelText="url"
-          id={'article-url'}
-          ref={(ref) => this.urlField = ref} // eslint-disable-line
-        /><br />
-        <TextField
           defaultValue={this.props.article.get('title') || this.state.currentArticle.get('title')}
-          disabled={this.props.article.get('state')}
+          disabled={this.props.article.get('id')}
           floatingLabelText="title"
-          id={'article-title'}
+          id="article-title"
           ref={(ref) => this.titleField = ref} // eslint-disable-line
         /><br />
         <Toggle
-          id={'remember-article'}
-          label={l('Remember article')}
+          id="remember-article"
+          label={this.rememberArticleInProgress() ? this.spinner('Remember article') : 'Remember article'}
```

Comment:
_Internationalization_ (i18n). 

However, I don't want to use the tool which is not configured. Better to create a separate issue and figure out how it works and do a proper internationalization.